test/test_dispatch.py

```diff
@@ -888,6 +888,9 @@ def test_defaultbackend_math(self):
 
     # TODO(jwtan): Use a C++ extension, like msnpu, to introduce a dangling impl and examine the output.
     def test_find_dangling_impls(self):
+        dangling_impls = C._dispatch_find_dangling_impls()
+        if len(dangling_impls) > 0:
+            print(dangling_impls)
         self.assertEqual(0, len(C._dispatch_find_dangling_impls()))
```

Comment:
nit: you can actually modify the asserEqual(msg=) to stringify the output for the assert failure
```suggestion
        self.assertEqual(0, len(C._dispatch_find_dangling_impls()), msg=f'Expect zero dangling impls, but found: {dangling_impls}')
```